autosklearn/pipeline/components/regression/mlp.py

```diff
@@ -0,0 +1,239 @@
+        """
+        Set n_iter=2 for the same reason as for SGD
+        """
+        from sklearn.neural_network import MLPRegressor
+        import sklearn.preprocessing
+        n_iter = max(n_iter, 2)
+
+        if refit:
+            self.estimator = None
+            self.scaler = None
+
+        if self.estimator is None:
+            self._fully_fit = False
+
+            self.max_iter = int(self.max_iter)
+            self.hidden_layer_depth = int(self.hidden_layer_depth)
+            self.num_nodes_per_layer = int(self.num_nodes_per_layer)
+            self.hidden_layer_sizes = tuple(self.num_nodes_per_layer
+                                            for i in range(self.hidden_layer_depth))
+            self.activation = str(self.activation)
+            self.alpha = float(self.alpha)
+            self.learning_rate_init = float(self.learning_rate_init)
+            self.early_stopping = str(self.early_stopping)
+            self.early_stopping_ = str(self.early_stopping)
+            if self.early_stopping == "train":
+                self.validation_fraction = 0.0
+                self.tol = float(self.tol)
+                self.n_iter_no_change = int(self.n_iter_no_change)
+                self.early_stopping_val = False
+            elif self.early_stopping == "valid":
+                self.validation_fraction = float(self.validation_fraction)
+                self.tol = float(self.tol)
+                self.n_iter_no_change = int(self.n_iter_no_change)
+                self.early_stopping_val = True
+            else:
+                raise ValueError("Set early stopping to unknown value %s" % self.early_stopping)
+            # elif self.early_stopping == "off":
+            #     self.validation_fraction = 0
+            #     self.tol = 10000
+            #     self.n_iter_no_change = self.max_iter
+            #     self.early_stopping_val = False
+
+            self.solver = self.solver
+
+            try:
+                self.batch_size = int(self.batch_size)
+            except ValueError:
+                self.batch_size = str(self.batch_size)
+
+            self.shuffle = check_for_bool(self.shuffle)
+            self.beta_1 = float(self.beta_1)
+            self.beta_2 = float(self.beta_2)
+            self.epsilon = float(self.epsilon)
+            self.beta_1 = float(self.beta_1)
+
+            self.max_iter = int(self.max_iter)
+            self.verbose = int(self.verbose)
+
+            n_iter = int(np.ceil(n_iter))
+
+            # initial fit of only increment trees
+            self.estimator = MLPRegressor(
+                hidden_layer_sizes=self.hidden_layer_sizes,
+                activation=self.activation,
+                solver=self.solver,
+                alpha=self.alpha,
+                batch_size=self.batch_size,
+                learning_rate_init=self.learning_rate_init,
+                max_iter=n_iter,
+                shuffle=self.shuffle,
+                random_state=self.random_state,
```

Comment:
There's a copy.copy() for classification around self.random_state. As it isn't here, maybe it's not necessary for classification?

Reply:
This is needed for the idempotent test, which only exists for classification. Fitting the MLP changes self.random_state, so subsequent calls to fit will use a different weight initialization. Copying the random_state guarantees that fit() will always do the same